assets/embed_gzip.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package assets
+
+import (
+	"compress/gzip"
+	"embed"
+	"io"
+	"io/fs"
+	"time"
+)
+
+const (
+	zipSuffix = ".gz"
+)
+
+type FileSystem struct {
+	embed embed.FS
+}
+
+func New(fs embed.FS) FileSystem {
+	return FileSystem{fs}
+}
+
+func (compressed FileSystem) Open(path string) (fs.File, error) {
+	// if we have the file in our embed FS, just return that (could be a
+	// dir)
```

Comment:
```suggestion
	// dir.)
```
let's have that comment in one line.